### PR TITLE
edit tests and bash for accounts_tmout

### PR DESCRIPTION
**Pull request by cyarbrough76**
#### Description:

- _Edited path in acounts_tmout for shared.sh and tests._

### CODE REVIEW

**Comment by mildas** [comment]
**Changes identified:**
Rule accounts_tmout:
 Found change in bash remediation.

**Recommended tests to execute:**
 build_product ol7
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-ol7-ds.xml accounts_tmout

**Comment by redhatrises** [comment]
Only problem with this change, is that it needs to check and remediate both `/etc/profile` AND `/etc/profile.d`

**Comment by vojtapolasek** [comment]
As I see it, you are moving the remediation from /etc/profile into /etc/profile.d/tmout.sh. What is the motivation behind this change? Change in STIG?

**Comment by thenefield** [comment]
> Only problem with this change, is that it needs to check and remediate both `/etc/profile` AND `/etc/profile.d`

The STIG restricts the check to just /etc/profile.d/*

Why would we apply the remediation in both places?

**Comment by thenefield** [comment]
> As I see it, you are moving the remediation from /etc/profile into /etc/profile.d/tmout.sh. What is the motivation behind this change? Change in STIG?

This is what the STIG and associated benchmark specifically look for.

**Comment by vojtapolasek** [comment]
One more thing. Please also update the rule description and ocil to reflect the change. Thank you.

**Comment by mildas** [comment]
**Changes identified:**
Rule accounts_tmout:
 Found change in bash remediation.

**Recommended tests to execute:**
 build_product ol7
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-ol7-ds.xml accounts_tmout

**Comment by mildas** [comment]
**Changes identified:**
Rule accounts_tmout:
 Found change in bash remediation.

**Recommended tests to execute:**
 build_product ol7
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-ol7-ds.xml accounts_tmout

**Comment by redhatrises** [comment]
This change needs to handle both `/etc/profile` AND `/etc/profile.d/`. You can overwrite the configuration in `/etc/profile.d/` by appending TMOUT to /etc/profile bringing the system out of compliance.

**Comment by vojtapolasek** [comment]
> This change needs to handle both `/etc/profile` AND `/etc/profile.d/`. You can overwrite the configuration in `/etc/profile.d/` by appending TMOUT to /etc/profile bringing the system out of compliance.

You can't change the variable by appending to /etc/profile, because the variable in /etc/profile.d/tmout.sh is made readonly.
You would need to define the variable and make it readonly before the point where files from /etc/profile.d directory are included.
Are we going to check also for this condition?

File: linux_os/guide/system/accounts/accounts-session/accounts_tmout/rule.yml [hunk 1]
```diff
@@ -5,9 +5,9 @@ prodtype: fedora,ol7,ol8,rhel6,rhel7,rhel8,rhv4,sle15,wrlinux1019
 title: 'Set Interactive Session Timeout'
 
 description: |-
-    Setting the <tt>TMOUT</tt> option in <tt>/etc/profile</tt> ensures that
+    Setting the <tt>TMOUT</tt> option in <tt>/etc/profile.d/tmout.sh</tt> ensures that
     all user sessions will terminate based on inactivity. The <tt>TMOUT</tt>
-    setting in <tt>/etc/profile</tt> should read as follows:
+    setting in <tt>/etc/profile.d/tmout.sh</tt> should read as follows:
     <pre>TMOUT=<sub idref="var_accounts_tmout" /></pre>
```
Comment (vojtapolasek) [inline linux_os/guide/system/accounts/accounts-session/accounts_tmout/rule.yml:11]: Please put here all three lines used in remediation, as I think that the part making the variable readonly is important.

**Comment by vojtapolasek** [comment]
It seems that you have committed some backup file, please remove it. Thank you.

File: linux_os/guide/system/accounts/accounts-session/accounts_tmout/rule.yml (hunk 1 above)
Comment (redhatrises) [reply linux_os/guide/system/accounts/accounts-session/accounts_tmout/rule.yml:11]: @vojtapolasek Making it readonly is a new rule and check, and it needs to go there and not in this rule.

Comment (vojtapolasek) [reply linux_os/guide/system/accounts/accounts-session/accounts_tmout/rule.yml:11]: My understanding is that this rule ensures that the TMOUT env variable is configured to some predefined value. The way proposed in latest STIG for rhel7 and also in this pull request ensures it very well, the readonly line makes sure that it is not trivial to overwrite the variable. I find this desirable. What is the reason to separate this into multiple rules?

**Comment by mildas** [comment]
**Changes identified:**
Rule accounts_tmout:
 Found change in bash remediation.

**Recommended tests to execute:**
 build_product ol7
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-ol7-ds.xml accounts_tmout

File: linux_os/guide/system/accounts/accounts-session/accounts_tmout/rule.yml (hunk 1 above)
Comment (redhatrises) [reply linux_os/guide/system/accounts/accounts-session/accounts_tmout/rule.yml:11]: Better support in operational environments which as the vendor we are moving towards. And actually, changing this to a one-liner like `declare -xr TMOUT=600` would be way better. DISA was in favor of this alternative.

Comment (vojtapolasek) [reply linux_os/guide/system/accounts/accounts-session/accounts_tmout/rule.yml:11]: I am okay with changing the line just to declare -xrj TMOUT= something. @cyarbrough76  @thenefield  is that OK for you? I also suggest that both /etc/profile and /etc/profile.d/tmout.sh are checked, while the remediation ensures that the line exists ONLY in /etc/profile.d/tmout.sh.

File: linux_os/guide/system/accounts/accounts-session/accounts_tmout/bash/shared.sh [hunk 2]
```diff
@@ -2,9 +2,14 @@
 . /usr/share/scap-security-guide/remediation_functions
 populate var_accounts_tmout
 
-if grep --silent ^TMOUT /etc/profile ; then
-        sed -i "s/^TMOUT.*/TMOUT=$var_accounts_tmout/g" /etc/profile
+if grep grep -q "^TMOUT=" /etc/profile.d/tmout.sh ; then
```
Comment (vojtapolasek) [inline linux_os/guide/system/accounts/accounts-session/accounts_tmout/bash/shared.sh:5]:
```suggestion
if grep -q "^TMOUT=" /etc/profile.d/tmout.sh ; then
```

File: linux_os/guide/system/accounts/accounts-session/accounts_tmout/bash/shared.sh [hunk 3]
```diff
@@ -2,9 +2,14 @@
...
+        sed -i "s/^TMOUT.*/TMOUT=$var_accounts_tmout/g" /etc/profile.d/tmout.sh
```
Comment (vojtapolasek) [inline linux_os/guide/system/accounts/accounts-session/accounts_tmout/bash/shared.sh:6]:
```suggestion
        sed -i "s/^TMOUT.*/declare -xr TMOUT=$var_accounts_tmout/g" /etc/profile.d/tmout.sh
```

File: linux_os/guide/system/accounts/accounts-session/accounts_tmout/bash/shared.sh [hunk 4]
```diff
@@ -2,9 +2,14 @@
...
 else
-        echo -e "\n# Set TMOUT to $var_accounts_tmout per security requirements" >> /etc/profile
-        echo "TMOUT=$var_accounts_tmout" >> /etc/profile
+        echo -e "\n# Set TMOUT to $var_accounts_tmout per security requirements" >> /etc/profile.d/tmout.sh
+        echo "TMOUT=$var_accounts_tmout" >> /etc/profile.d/tmout.sh
```
Comment (vojtapolasek) [inline linux_os/guide/system/accounts/accounts-session/accounts_tmout/bash/shared.sh:9]:
```suggestion
        echo "declare -xr TMOUT=$var_accounts_tmout" >> /etc/profile.d/tmout.sh
```

File: linux_os/guide/system/accounts/accounts-session/accounts_tmout/bash/shared.sh [hunk 5]
```diff
@@ -2,9 +2,14 @@
...
+        echo "readonly TMOUT" >> /etc/profile.d/tmout.sh
+        echo "export TMOUT" >> /etc/profile.d/tmout.sh
+fi
+if grep grep -q "^TMOUT=" /etc/profile ; then
```
Comment (vojtapolasek) [inline linux_os/guide/system/accounts/accounts-session/accounts_tmout/bash/shared.sh:13]:
```suggestion
if grep -q "^TMOUT=" /etc/profile ; then
```

**Comment by matejak** [comment]
I am closing this, as the PR
- is incomplete: The bash remediation is not complemented by OVAL or Ansible updates, and
- needs clarification: The readonly/export of the env variable has to be either agreed upon to put into one rule, or it has to be split into multiple rules that are complement each other.

As the PR author hasn't responded to reviewer feedback for a long time, it is appropriate to close the PR. Please reopen if you intend to continue its development.

**Pull request closed without merging**